app/components/primer/popover_component.html.erb

```diff
@@ -0,0 +1,19 @@
+<%= render Primer::BaseComponent.new(**@kwargs) do %>
+  <% if message %>
+    <%= render message.component do %>
```

Comment:
I thought this might be a nice way of keeping the Message slot's `kwargs` private to itself, instead of exposing them via `attr_reader :kwargs`, which felt a bit dirty. However, my test fails with:

```
PrimerPopoverComponentTest#test_renders_default_styling:
NoMethodError: undefined method `component' for #<ViewComponent::Slot:0x00007f85f522d738 @content="">
Did you mean?  content
    test/components/popover_component_test.rb:9:in `test_renders_default_styling'
```

Reply:
I thinkyou need to declare the class you are using on the `with_slot` declaration

```rb
with_slot :message, class_name: 'Message'
with_slot :heading, class_name: 'Heading'
...
```

Reply:
Ohh gotcha. The readme example made me think the class declaration was only necessary if you weren’t using the expected name.